extensions/bootstrap/ButtonDropdown.php

```diff
@@ -79,11 +79,13 @@ public function run()
         $options = $this->containerOptions;
         $tag = ArrayHelper::remove($options, 'tag', 'div');
 
-        echo Html::beginTag($tag, $this->containerOptions);
-        echo "\n" . $this->renderButton();
-        echo "\n" . $this->renderDropdown();
-        echo "\n" . Html::endTag($tag);
         $this->registerPlugin('button');
+        return implode("\n", [
```

Comment:
honest question. Why is this required? the code generated here is not to be debuged or read by humans since is dinamically generated.